draft-ietf-quic-load-balancers.md

```diff
@@ -444,71 +444,157 @@ routing information of any given 4-tuple.
-the property that each encrypted nonce value depends on all server ID bits, and
-each encrypted server ID bit depends on all nonce bits and all server ID bits.
-This mitigates attacks against stream ciphers in which attackers simply flip
-encrypted server-ID bits.
+4. XOR the most significant bits of the ciphertext with right_0 to form
+right_1.
 
-The output of the decryption is the server ID that the load balancer uses for
-routing.
+    Thus steps 3 and 4 can be expressed as
 
-### Server Actions
+    ```
+    right_1 = right_0 ^ truncate(AES_ECB(key, expand(left_0, 0x01))
+    ```
 
-When generating a routable connection ID, the server writes arbitrary bits into
-its nonce octets, and its provided server ID into the server ID octets. See
-{{cid-entropy}} for nonce generation considerations.
+5. Repeat steps 3 and 4, but use them to compute left_1 by expanding and
+encrypting right_1 with the least significant octet as 0x02 and XOR the
+results with left_0.
+
+    ```
+    left_1 = left_0 ^ truncate(AES_ECB(key, expand(right_1), 0x02))
+    ```
+
+6. Repeat steps 3 and 4, but use them to compute right_2 by expanding and
+encrypting left_1 with the least significant octet as 0x03 and XOR the
+results with right_1.
+
+    ```
+    right_2 = right_1 ^ truncate(AES_ECB(key, expand(left_1, 0x03))
+    ```
+
+7. Repeat steps 3 and 4, but use them to compute left_2 by expanding and
+encrypting right_2 with the least significant octet as 0x04 and XOR the
+results with left_1.
+
+    ```
+    left_2 = left_1 ^ truncate(AES_ECB(key, expand(right_2), 0x04))
+    ```
+
+8. The server concatenates left_2 with right_2 to form the ciphertext CID,
+which it appends to the first octet.
+
+The following example executes the steps for the provided inputs. Note that the
+plaintext is of odd octet length, so the middle octet will be split evenly
+left_0 and right_0.
+
+~~~pseudocode
+server_id = 0x3144a
+nonce = 0x9c69c275
+key = 0xfdf726a9893ec05c0632d30z6680baf0
+
+// step 1
+plaintext_CID = 0x31441a9c69c275
 
-The server encrypts the server ID using exactly the algorithm as described in
-{{encrypted-short-load-balancer-actions}}, performing the three passes
-in reverse order.
+// step 2
+left_0 = 0x31441a9
+right_0 = 0xc69c275
+
+// step 3
+aes_input = 0x31441a90000000000000000000000001
+ciphertext = 0xdea73834473e88afee51be7f6bdff0e7
+
+// step 4
+right_1 = 0xc69c275 ^ 0xdea7383 = 0x183b1f6
+
+// step 5
+aes_input = 0x183b1f60000000000000000000000002
+aes_output = 0x15ab4a6f252c0283a0446c74c3f98860
+left_1 = 0x31441a9 ^ 0x15ab4a6 = 0x24ef50f
+
+// step 6
+AES input = 0x24ef50f0000000000000000000000003
+AES output = 0xbeaca161e903ebb97cfda599a29ad8ff
+right_2 = 0x183b1f6 ^ 0xbeaca16 = 0xa697be0
+
+// step 7
+AES input: = 0xa697be00000000000000000000000004
+AES output = 0x13ea04a5e3c707bf197e8fcbcd43ef98
+left_2 = 0x24ef50f ^ 0x13ea04a = 0x3705545
+
+// step 8
+cid = first_octet || left_2 || right_2 = 0x073705545a697be0
+~~~
+
```

Comment:
Consider adding example with an uneven break, such as 3 bytes server ID and 4 bytes nonce. This might save us lots of trouble!

Reply:
That what the current example is!

Reply:
Yes, I missed that when I first read it. Sorry.